azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -433,14 +437,32 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
           jobProps.put(key, clusterSpecificProps.get(key));
         }
       }
+
+      // Override any plugin load props if specified.
+      Props pluginLoadPropsCopy = pluginLoadProps;
```

Comment:
Why not create clone here itself? 
Props pluginLoadPropsCopy = Props.clone(pluginLoadProps);

Reply:
Why clone unnecessarily when it is not needed? That is why cloning is done when if condition is true.

Reply:
For safety. Later on by mistake if anyone adds anything in Copy before you clone in future.

Reply:
Sure, I was trying to keep it efficient but safety trumps it.